portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -57,28 +63,37 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
-
-    String userChoice = request.getParameter("num");
+    Translate translate = TranslateOptions.getDefaultInstance().getService();
+    
+    String numChoice = request.getParameter("num");
+    String languageChoice = request.getParameter("language");
+    Translate.TranslateOption targetLanguage = Translate.TranslateOption.targetLanguage(languageChoice);
     int maxComments;
-    maxComments = Integer.parseInt(userChoice);
-    int i =0;
-    for (Entity entity : results.asIterable()) {
-        if (i < maxComments){
+    maxComments = Integer.parseInt(numChoice);
```

Comment:
Since the num choice is sent by the user and could be set to a non-number, you might want to `try { } catch ...` it and explicitly set the response status code or use a default value if the number is invalid.

Reply:
added it